From a0aa90a0d7d662d9e1528880bebf4650a298d237 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Thu, 21 Dec 2017 10:54:06 +0800
Subject: [PATCH] arm64: dts: rockchip: add basic dts for RK3308 FPGA platform

The FPGA platform is used for rk3308 validation, which has
one cortex-a35 cpu and uart/timer peripherals.

Change-Id: I3a78cc5afec04d954f4d2715b6a6066400a1c513
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/Makefile        |  1 +
 arch/arm64/boot/dts/rockchip/rk3308-fpga.dts | 46 ++++++++++++++++++++
 2 files changed, 47 insertions(+)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3308-fpga.dts

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index b464ee609d26..f28ba7f23746 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -1,6 +1,7 @@
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-evb-ddr3-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-evb-ddr3-lvds-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-evb-ddr4-v10.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3308-fpga.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-evb-lp3-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-863-lp3-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3328-evb.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-fpga.dts b/arch/arm64/boot/dts/rockchip/rk3308-fpga.dts
new file mode 100644
index 000000000000..1aae32a2b238
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3308-fpga.dts
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2017 Fuzhou Rockchip Electronics Co., Ltd
+ *
+ * SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+ */
+
+/dts-v1/;
+
+#include "rk3308.dtsi"
+
+/ {
+	model = "Rockchip RK3308 FPGA Platform";
+	compatible = "rockchip,rk3308-fpga", "rockchip,rk3308";
+
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xff0b0000 console=ttyFIQ0 init=/init initrd=0x9000000,0x18bfc0";
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x00200000 0x0 0x0FE00000>;
+	};
+
+	fiq_debugger: fiq-debugger {
+		compatible = "rockchip,fiq-debugger";
+		rockchip,serial-id = <1>;
+		rockchip,wake-irq = <0>;
+		rockchip,irq-mode-enable = <1>;
+		rockchip,baudrate = <1500000>;  /* Only 115200 and 1500000 */
+		interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH 0>;
+		status = "ok";
+	};
+
+};
+
+&cpu1 {
+	/delete-property/enable-method;
+};
+
+&cpu2 {
+	/delete-property/enable-method;
+};
+
+&cpu3 {
+	/delete-property/enable-method;
+};
-- 
2.35.3

